frontend/pages/code/[id]/chat.tsx

```diff
@@ -0,0 +1,141 @@
+import { FC, RefObject, useEffect, useRef, useState } from 'react'
+
+import { useSession } from 'next-auth/react'
+import { useRouter } from 'next/router'
+import * as socketIO from 'socket.io-client'
+import { getChatHistory } from '@/services/collaboration-service-api'
+
+interface ICollaborator {
+    name: string
+}
+
+const formatTimestamp = (timestamp: string) => {
+    const date = new Date(timestamp)
+    return date.toLocaleTimeString([], { hour: '2-digit', minute: '2-digit', hour12: true }).toUpperCase()
+}
+
+export interface IMessage {
+    senderId: string
+    message: string
+    createdAt: Date
+    roomId: string
+}
```

Comment:
can move this to types folder as well, i believe theres a file with collaboration api, so can paste into it.